clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/provider/agent/TaskHealthCachingAgent.java

```diff
@@ -129,34 +129,38 @@ public TaskHealthCachingAgent(
             Keys.getTaskDefinitionKey(accountName, region, service.getTaskDefinition());
         TaskDefinition taskDefinition = taskDefinitionCacheClient.get(taskDefinitionCacheKey);
 
-        if (isContainerMissingNetworking(task)) {
+        if (taskDefinition == null) {
           log.debug(
-              "Task '{}' is missing networking. Will not retrieve health.", task.getTaskArn());
+              "Provided task definition '{}' is null for service '{}'",
+              service.getTaskDefinition(),
+              service.getServiceName());
+          continue;
+        }
+
+        boolean lacksNetworkBindings = isTaskMissingNetworkBindings(task);
+        if (task.getContainers().isEmpty()
+            || (lacksNetworkBindings && isTaskMissingNetworkInterfaces(task))) {
+          log.debug(
+              "Task '{}' is missing networking. Will not retrieve load balancer health.",
+              task.getTaskArn());
           continue;
         }
 
         TaskHealth taskHealth = null;
-        Collection<Container> containers = task.getContainers();
-
-        for (Container container : containers) {
-          if (container.getNetworkBindings().size() >= 1) {
-            taskHealth =
-                inferHealthNetworkBindedContainer(
-                    targetHealthCacheClient,
-                    task,
-                    containerInstance,
-                    serviceName,
-                    service,
-                    taskDefinition);
-            break;
-          }
+        // ideally, could determine health check method by looking at taskDef.networkMode,
+        // however this isn't reliably cached yet, so reusing network binding check.
+        if (!lacksNetworkBindings) {
+          taskHealth =
+              inferHealthNetworkBindedContainer(
+                  targetHealthCacheClient, task, containerInstance, serviceName, service);
         }
 
         if (taskHealth == null) {
```

Comment:
We should probably update this to be an else if condition since you can't have both networking bindings and interfaces present 

`else if (!lacksNetworkInterfaces) {`

Reply:
I wanted to keep the existing logic (outside removing the loop) mostly intact here, but I agree it's beneficial to log if both are found (see above). Maybe I can make this & the logging change in a separate PR so they're isolated?

Reply:
Sounds good to me!